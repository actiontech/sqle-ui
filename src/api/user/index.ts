/* tslint:disable no-identical-functions */
/* tslint:disable no-useless-cast */
/* tslint:disable no-unnecessary-type-assertion */
/* tslint:disable no-big-function  */
/* tslint:disable no-duplicate-string  */
import ServiceBase from '../Service.base';
import { AxiosRequestConfig } from 'axios';

import {
  ILoginV1Params,
  ILoginV1Return,
  IGetCurrentUserV1Return,
  IUpdateCurrentUserV1Params,
  IUpdateCurrentUserV1Return,
  IUpdateCurrentUserPasswordV1Params,
  IUpdateCurrentUserPasswordV1Return,
  IGetUserTipListV1Return,
  IGetUserListV1Params,
  IGetUserListV1Return,
  ICreateUserV1Params,
  ICreateUserV1Return,
  IGetUserV1Params,
  IGetUserV1Return,
  IDeleteUserV1Params,
  IDeleteUserV1Return,
  IUpdateUserV1Params,
  IUpdateUserV1Return,
  IUpdateOtherUserPasswordV1Params,
  IUpdateOtherUserPasswordV1Return
} from './index.d';

class UserService extends ServiceBase {
  public loginV1(params: ILoginV1Params, options?: AxiosRequestConfig) {
    const paramsData = this.cloneDeep(params);
    return this.post<ILoginV1Return>('/v1/login', paramsData, options);
  }

  public getCurrentUserV1(options?: AxiosRequestConfig) {
    return this.get<IGetCurrentUserV1Return>('/v1/user', undefined, options);
  }

  public updateCurrentUserV1(
    params: IUpdateCurrentUserV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    return this.patch<IUpdateCurrentUserV1Return>(
      '/v1/user',
      paramsData,
      options
    );
  }

  public UpdateCurrentUserPasswordV1(
    params: IUpdateCurrentUserPasswordV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    return this.put<IUpdateCurrentUserPasswordV1Return>(
      '/v1/user/password',
      paramsData,
      options
    );
  }

  public getUserTipListV1(options?: AxiosRequestConfig) {
    return this.get<IGetUserTipListV1Return>(
      '/v1/user_tips',
      undefined,
      options
    );
  }

  public getUserListV1(
    params: IGetUserListV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    return this.get<IGetUserListV1Return>('/v1/users', paramsData, options);
  }

  public createUserV1(
    params: ICreateUserV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    return this.post<ICreateUserV1Return>('/v1/users', paramsData, options);
  }

  public getUserV1(params: IGetUserV1Params, options?: AxiosRequestConfig) {
    const paramsData = this.cloneDeep(params);
    const user_name = paramsData.user_name;
    delete paramsData.user_name;

    return this.get<IGetUserV1Return>(
      `/v1/users/${user_name}/`,
      paramsData,
      options
    );
  }

  public deleteUserV1(
    params: IDeleteUserV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    const user_name = paramsData.user_name;
    delete paramsData.user_name;

    return this.delete<IDeleteUserV1Return>(
      `/v1/users/${user_name}/`,
      paramsData,
      options
    );
  }

  public updateUserV1(
    params: IUpdateUserV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    const user_name = paramsData.user_name;
    delete paramsData.user_name;

    return this.patch<IUpdateUserV1Return>(
      `/v1/users/${user_name}/`,
      paramsData,
      options
    );
  }

  public UpdateOtherUserPasswordV1(
    params: IUpdateOtherUserPasswordV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    const user_name = paramsData.user_name;
    delete paramsData.user_name;

    return this.patch<IUpdateOtherUserPasswordV1Return>(
      `/v1/users/${user_name}/password`,
      paramsData,
      options
    );
  }
}

export default new UserService();
