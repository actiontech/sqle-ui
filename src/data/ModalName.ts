export enum ModalName {
  // user manage
  Add_Role = 'ADD_ROLE',
  Update_Role = 'UPDATE_ROLE',

  Add_User = 'ADD_USER',
  Update_User = 'Update_User',
  Update_User_Password = 'Update_User_Password',

  // whitelist manage
  Add_Whitelist = 'ADD_WHITELIST',
  Update_Whitelist = 'UPDATE_WHITELIST',

  // order
  Modify_Sql = 'Modify_Sql',

  // account
  Modify_Password = 'Modify_Password',

  // rule template
  Clone_Rule_Template = 'CLONE_RULE_TEMPLATE',

  // header
  SHOW_VERSION = 'SHOW_VERSION',
}
