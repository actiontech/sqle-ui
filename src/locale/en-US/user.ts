// eslint-disable-next-line import/no-anonymous-default-export
export default {
  pageTitle: 'User Manage',
  pageDesc:
    'You can manage the users of the platform here, and users can only see or operate the data sources contained in all the roles they have bound.',

  userListTitle: 'User List',

  createUser: {
    button: 'Create User',
    modalTitle: 'Create User',
    createSuccessTips: 'Create User "{{name}}" successfully',
  },

  updateUser: {
    modalTitle: 'Update User',
    updateSuccessTips: 'Update user "{{name}}" successfully',
  },

  userForm: {
    username: 'Username',
    password: 'Password',
    passwordConfirm: 'Confirm Password',
    email: 'Email',
    role: 'Role',

    passwordConfirmPlaceholder: 'Please keep the same password twice',
  },

  userListFilter: {
    usernamePlaceholder: 'Search username',
    rolePlaceholder: 'Search role',
  },

  deleteUser: {
    confirmTitle: 'Are you sure delete this user: "{{username}}"?',
    deleting: 'Deleting user: "{{username}}...',
    deleteSuccess: 'Delete user "{{username}}" successfully',
  },

  userList: {
    emailPlaceholder: 'Empty',
    rolePlaceHolder: 'Empty',
    role: 'Role',
  },

  roleListTitle: 'Role List',

  createRole: {
    button: 'Create Role',
    modalTitle: 'Create Role',
    createSuccessTips: 'Create role "{{name}}" successfully',
  },

  updateRole: {
    modalTitle: 'Update Role',
    updateSuccessTips: 'Update role "{{name}}" successfully',
  },

  deleteRole: {
    deleteTips: 'Are you sure delete this role: "{{name}}"?',
    deleting: 'Deleting role "{{name}}"...',
    deleteSuccessTips: 'Delete Role "{{name}}" successfully',
  },

  roleForm: {
    roleName: 'Role Name',
    roleDesc: 'Role Describe',
    databases: 'Bind Databases',
    usernames: 'Bind Usernames',
  },

  roleListFilter: {
    usernamePlaceholder: 'Search username',
    rolePlaceholder: 'Search role',
    databasePlaceholder: 'Search database',
  },

  roleList: {
    roleDescPlaceholder: 'Empty',
    usernamePlaceholder: 'Empty',
    databasePlaceholder: 'Empty',
    username: 'Usernames',
    database: 'Databases',
  },
};
