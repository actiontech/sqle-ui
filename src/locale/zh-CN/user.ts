// eslint-disable-next-line import/no-anonymous-default-export
export default {
  pageTitle: '用户管理',
  pageDesc:
    '你可以在这里管理平台的用户，用户只能看到或操作其已绑定的所有角色中包含的数据源，和其所属的用户组所拥有的所有权限。',
  userListTitle: '用户列表',

  createUser: {
    button: '创建用户',
    modalTitle: '创建用户',
    createSuccessTips: '创建用户 "{{name}}" 成功',
  },

  updateUser: {
    modalTitle: '更新用户',
    updateSuccessTips: '更新用户 "{{name}}" 成功',
  },

  updateUserPassword: {
    button: '修改该用户密码',
    title: '修改{{name}}用户密码',

    successTips: '修改{{name}}用户密码成功',
  },

  table: {
    userType: '用户类型',
    status: '用户状态',
  },

  userState: {
    normal: '正常',
    disabled: '已禁用',
  },

  userForm: {
    username: '用户名',
    password: '密码',
    passwordConfirm: '确认密码',
    email: '邮箱',
    role: '角色',
    disabled: '禁用用户',
    userGroup: '所属用户组',
    wechat: '微信id',

    passwordConfirmPlaceholder: '请保持两次密码输入一致',
  },

  userListFilter: {
    usernamePlaceholder: '请输入要搜索的用户名',
    rolePlaceholder: '请输入要搜索的角色',
  },

  deleteUser: {
    confirmTitle: '确认要删除用户: "{{username}}"?',
    deleting: '正在删除用户: "{{username}}...',
    deleteSuccess: '删除用户 "{{username}}" 成功',
  },

  userList: {
    emailPlaceholder: '未填写邮箱',
    rolePlaceHolder: '未拥有任何角色',
    role: '拥有角色',
  },
};
