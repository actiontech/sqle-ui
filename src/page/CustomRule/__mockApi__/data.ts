import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
