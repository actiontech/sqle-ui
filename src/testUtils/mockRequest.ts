import { AxiosResponse } from 'axios';
import instance from '../api/instance';
import role from '../api/role';
import ruleTemplate from '../api/rule_template';
import user from '../api/user';
import workflow from '../api/workflow';

export const successData = (data: any, otherData?: any) => {
  return {
    code: 0,
    message: '',
    data,
    ...otherData,
  };
};

export const failedData = (data?: any) => {
  return {
    code: 100,
    message: 'error',
    data,
  };
};

export const resolveImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.resolve({
    status,
    headers,
    config,
    statusText,
    data: successData(data),
  });
};

export const rejectImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.reject({
    status,
    headers,
    config,
    statusText,
    data: failedData(data),
  });
};

export const resolveThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {},
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: successData(data, otherData),
      });
    }, 3000);
  });
};

export const resolveErrorThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const rejectThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((_, rej) => {
    setTimeout(() => {
      rej({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const throwErrorThreeSecond = (
  error = 'error',
  { status = 500, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise((res) => {
    throw new Error(error);
  });
};

export const mockUseInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ instance_name: 'instance1' }])
  );
  return spy;
};

export const mockUseInstanceSchema = () => {
  const spy = jest.spyOn(instance, 'getInstanceSchemasV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({ schema_name_list: ['schema1'] })
  );
  return spy;
};

export const mockUseRole = () => {
  const spy = jest.spyOn(role, 'getRoleTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ role_name: 'role_name1' }])
  );
  return spy;
};

export const mockUseRuleTemplate = () => {
  const spy = jest.spyOn(ruleTemplate, 'getRuleTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ rule_template_name: 'rule_template_name1' }])
  );
  return spy;
};

export const mockUseUsername = () => {
  const spy = jest.spyOn(user, 'getUserTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ user_name: 'user_name1' }])
  );
  return spy;
};

export const mockUseWorkflowTemplate = () => {
  const spy = jest.spyOn(workflow, 'getWorkflowTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ workflow_template_name: 'workflow-template-name-1' }])
  );
  return spy;
};
