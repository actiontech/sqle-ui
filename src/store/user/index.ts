import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { SystemRole } from '../../data/common';
import StorageKey from '../../data/StorageKey';
import { SupportTheme } from '../../theme';
import LocalStorageWrapper from '../../utils/LocalStorageWrapper';

type UserReduxState = {
  username: string;
  role: SystemRole | '';
  token: string;
  theme: string;
};

const initialState: UserReduxState = {
  username: '',
  role: '',
  token: LocalStorageWrapper.getOrDefault(StorageKey.Token, ''),
  theme: LocalStorageWrapper.getOrDefault(StorageKey.Theme, SupportTheme.LIGHT),
};

const user = createSlice({
  name: 'user',
  initialState,
  reducers: {
    updateUser: (
      state,
      {
        payload: { username, role },
      }: PayloadAction<{ username: string; role: SystemRole | '' }>
    ) => {
      state.username = username;
      state.role = role;
      LocalStorageWrapper.set(StorageKey.Token, state.token);
    },
    updateTheme: (
      state,
      { payload: { theme } }: PayloadAction<{ theme: SupportTheme }>
    ) => {
      state.theme = theme;
      LocalStorageWrapper.set(StorageKey.Theme, theme);
    },
    updateToken: (
      state,
      { payload: { token } }: PayloadAction<{ token: string }>
    ) => {
      state.token = token;
    },
  },
});

export const { updateUser, updateTheme, updateToken } = user.actions;

export default user.reducer;
