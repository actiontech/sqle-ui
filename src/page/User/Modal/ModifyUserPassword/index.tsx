import { useBoolean } from 'ahooks';
import { Button, Form, Input, message } from 'antd';
import { useForm } from 'antd/lib/form/Form';
import Modal from 'antd/lib/modal/Modal';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';
import { IUserResV1 } from '../../../../api/common';
import user from '../../../../api/user';
import { ModalFormLayout, ResponseCode } from '../../../../data/common';
import { ModalName } from '../../../../data/ModalName';
import { IReduxState } from '../../../../store';
import { updateUserManageModalStatus } from '../../../../store/userManage';

const ModifyUserPassword = () => {
  const visible = useSelector<IReduxState, boolean>(
    (state) => state.userManage.modalStatus[ModalName.Update_User_Password]
  );
  const selectUser = useSelector<IReduxState, IUserResV1 | null>(
    (state) => state.userManage.selectUser
  );
  const [form] = useForm<{ password: string; passwordConfirm: string }>();
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const [submitLoading, { setTrue: startSubmit, setFalse: submitFinish }] =
    useBoolean();

  const close = () => {
    form.resetFields();
    dispatch(
      updateUserManageModalStatus({
        modalName: ModalName.Update_User_Password,
        status: false,
      })
    );
  };
  const submit = async () => {
    const value = await form.validateFields();
    startSubmit();
    user
      .UpdateOtherUserPasswordV1({
        user_name: selectUser?.user_name ?? '',
        password: value.password,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          message.success(t('user.updateUserPassword.successTips'));
          close();
        }
      })
      .finally(() => {
        submitFinish();
      });
  };

  return (
    <Modal
      visible={visible}
      title={t('user.updateUserPassword.title', {
        name: selectUser?.user_name,
      })}
      closable={false}
      footer={
        <>
          <Button onClick={close} disabled={submitLoading}>
            {t('common.close')}
          </Button>
          <Button type="primary" onClick={submit} loading={submitLoading}>
            {t('common.submit')}
          </Button>
        </>
      }
    >
      <Form {...ModalFormLayout} form={form}>
        <Form.Item
          name="password"
          label={t('user.userForm.password')}
          rules={[
            {
              required: true,
            },
          ]}
        >
          <Input.Password
            placeholder={t('common.form.placeholder.input', {
              name: t('user.userForm.password'),
            })}
          />
        </Form.Item>
        <Form.Item
          name="passwordConfirm"
          label={t('user.userForm.passwordConfirm')}
          dependencies={['password']}
          rules={[
            {
              required: true,
            },
            ({ getFieldValue }) => ({
              validator(_, value) {
                if (!value || getFieldValue('password') === value) {
                  return Promise.resolve();
                }
                return Promise.reject(
                  new Error(t('common.form.rule.passwordNotMatch'))
                );
              },
            }),
          ]}
        >
          <Input.Password
            placeholder={t('user.userForm.passwordConfirmPlaceholder')}
          />
        </Form.Item>
      </Form>
    </Modal>
  );
};

export default ModifyUserPassword;
