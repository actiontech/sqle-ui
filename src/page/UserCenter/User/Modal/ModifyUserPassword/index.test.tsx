import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import ModifyUserPassword from '.';
import user from '../../../../../api/user';
import {
  mockUseDispatch,
  mockUseSelector,
} from '../../../../../testUtils/mockRedux';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';
import { ModalName } from '../../../../../data/ModalName';

describe('User/ModifyUserPassword', () => {
  let dispatchSpy: jest.Mock;
  beforeEach(() => {
    jest.useFakeTimers();
    mockUseSelector({
      userManage: {
        modalStatus: { [ModalName.Update_User_Password]: true },
        selectUser: {
          user_name: 'root',
          email: 'user@123.com',
          role_name_list: ['role_name1'],
        },
      },
    });
    const { scopeDispatch } = mockUseDispatch();
    dispatchSpy = scopeDispatch;
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  const mockUpdateOtherUserPasswordV1 = () => {
    const spy = jest.spyOn(user, 'UpdateOtherUserPasswordV1');
    spy.mockImplementation(() => resolveThreeSecond({}));
    return spy;
  };

  test('should match snapshot', async () => {
    mockUseSelector({
      userManage: {
        modalStatus: { [ModalName.Update_User_Password]: false },
        selectUser: {
          user_name: 'root',
          email: 'user@123.com',
          role_name_list: ['role_name1'],
        },
      },
    });
    const { baseElement, rerender } = render(<ModifyUserPassword />);
    expect(baseElement).toMatchSnapshot();
    mockUseSelector({
      userManage: {
        modalStatus: { [ModalName.Update_User_Password]: true },
        selectUser: {
          user_name: 'root',
          email: 'user@123.com',
          role_name_list: ['role_name1'],
        },
      },
    });
    rerender(<ModifyUserPassword />);
    expect(baseElement).toMatchSnapshot();
  });

  test('should send modify password request when user input all fields', async () => {
    const updateSpy = mockUpdateOtherUserPasswordV1();
    render(<ModifyUserPassword />, undefined);

    fireEvent.input(screen.getByLabelText('user.userForm.password'), {
      target: { value: '222' },
    });
    fireEvent.input(screen.getByLabelText('user.userForm.passwordConfirm'), {
      target: { value: '222' },
    });
    fireEvent.click(screen.getByText('common.submit'));
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(updateSpy).toBeCalledTimes(1);
    expect(updateSpy).toBeCalledWith({
      password: '222',
      user_name: 'root',
    });
    expect(screen.getByText('common.submit').parentNode).toHaveClass(
      'ant-btn-loading'
    );
    expect(screen.getByText('common.close').parentNode).toHaveAttribute(
      'disabled'
    );
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(dispatchSpy).toBeCalledTimes(1);
    expect(dispatchSpy).toBeCalledWith({
      payload: { modalName: 'Update_User_Password', status: false },
      type: 'user/updateModalStatus',
    });
  });

  test('should reset all fields when user close modal', async () => {
    render(<ModifyUserPassword />);

    fireEvent.input(screen.getByLabelText('user.userForm.password'), {
      target: { value: '222' },
    });
    fireEvent.input(screen.getByLabelText('user.userForm.passwordConfirm'), {
      target: { value: '222' },
    });
    fireEvent.click(screen.getByText('common.close'));
    expect(dispatchSpy).toBeCalledWith({
      payload: { modalName: 'Update_User_Password', status: false },
      type: 'user/updateModalStatus',
    });
    expect(screen.getByLabelText('user.userForm.password')).toHaveValue('');
    expect(screen.getByLabelText('user.userForm.passwordConfirm')).toHaveValue(
      ''
    );
  });
});
