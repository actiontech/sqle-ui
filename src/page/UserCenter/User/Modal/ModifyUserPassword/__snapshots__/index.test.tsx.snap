// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User/ModifyUserPassword should match snapshot 1`] = `
<body>
  <div />
</body>
`;

exports[`User/ModifyUserPassword should match snapshot 2`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal"
          role="dialog"
          style="width: 520px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                user.updateUserPassword.title
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <form
                class="ant-form ant-form-horizontal"
              >
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                    >
                      <label
                        class="ant-form-item-required"
                        for="password"
                        title="user.userForm.password"
                      >
                        user.userForm.password
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <span
                            class="ant-input-affix-wrapper ant-input-password"
                          >
                            <input
                              aria-required="true"
                              class="ant-input"
                              id="password"
                              placeholder="common.form.placeholder.input"
                              type="password"
                              value=""
                            />
                            <span
                              class="ant-input-suffix"
                            >
                              <span
                                aria-label="eye-invisible"
                                class="anticon anticon-eye-invisible ant-input-password-icon"
                                role="img"
                                tabindex="-1"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="eye-invisible"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                  />
                                  <path
                                    d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item"
                >
                  <div
                    class="ant-row ant-form-item-row"
                  >
                    <div
                      class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
                    >
                      <label
                        class="ant-form-item-required"
                        for="passwordConfirm"
                        title="user.userForm.passwordConfirm"
                      >
                        user.userForm.passwordConfirm
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <span
                            class="ant-input-affix-wrapper ant-input-password"
                          >
                            <input
                              aria-required="true"
                              class="ant-input"
                              id="passwordConfirm"
                              placeholder="user.userForm.passwordConfirmPlaceholder"
                              type="password"
                              value=""
                            />
                            <span
                              class="ant-input-suffix"
                            >
                              <span
                                aria-label="eye-invisible"
                                class="anticon anticon-eye-invisible ant-input-password-icon"
                                role="img"
                                tabindex="-1"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="eye-invisible"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                  />
                                  <path
                                    d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="ant-modal-footer"
            >
              <button
                class="ant-btn ant-btn-default"
                type="button"
              >
                <span>
                  common.close
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  common.submit
                </span>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
